src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/RapidCatchUp.java

```diff
@@ -0,0 +1,223 @@
+package com.hms_networks.americas.sc.extensions.historicaldata;
+
+import com.ewon.ewonitf.Exporter;
+import com.hms_networks.americas.sc.extensions.datapoint.DataPoint;
+import com.hms_networks.americas.sc.extensions.json.JSONException;
+import com.hms_networks.americas.sc.extensions.string.StringUtils;
+import com.hms_networks.americas.sc.extensions.system.time.SCTimeUtils;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+/**
+ * This class addresses the need for historical data tracking to "catch up" quickly when there are
+ * periods where the tracker is not running. For example, the Flexy is powered off for several
+ * weeks. For correct behavior, it is required that injectJvmLocalTime from {@link SCTimeUtils} is
+ * called at startup.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ * @since 1.14.2
+ */
+public class RapidCatchUp {
+
+  /**
+   * Threshold for enabling rapid catch up mode in milliseconds. If current time - start time is
+   * less than this amount, catchup mode will not be enabled.
+   */
+  // 1000 * 60 * 60 = 3,600,000 milliseconds = 1 hour
+  public static final long RAPID_CATCH_UP_TIME_DELTA_MILLISECONDS = 1000 * 60 * 60;
+
+  /** Starting point for request period, which changes based on previous result. */
+  // 1000 * 60 * 10 = 600,0000 milliseconds = 10 minutes
+  public static final long RAPID_CATCH_UP_TIME_BASE = 1000 * 60 * 10;
+
+  /** Maximum amount of time for the request period. */
+  // 144 * 10 minutes = 24 hours
+  public static final long RAPID_CATCH_UP_TIME_DURATION_MAX = 144 * RAPID_CATCH_UP_TIME_BASE;
+
+  /** Dynamic duration for requesting historical data during catch up period */
+  public static long catchUpRequestDurationMilliseconds = RAPID_CATCH_UP_TIME_BASE;
+
+  /** Stream max read tries */
+  public static final int MAX_STREAM_READS = 3;
+
+  /** Stream buffer size */
+  public static final int STREAM_BUFFER_SIZE_BYTES = 256;
+
+  /**
+   * True/false enter rapid catch up mode. The decision is based on if the last read to historical
+   * data was empty and how far behind is the historical tracking time from current system time.
```

Comment:
```suggestion
   * True/False enter rapid catch up mode. The decision is based on if the last historical
   * data read was empty and how far behind is the historical tracking time from current system time.
```

Reply:
I think @alexjhawk's suggestion of {@code true}/{@code false} is preferred here.